$OpenBSD: patch-libhsm_src_lib_libhsm_c,v 1.1 2021/11/28 10:37:04 sthen Exp $

Don't assume that LDNS >= 1.7 means that ed25519/ed448 support is present

Index: libhsm/src/lib/libhsm.c
--- libhsm/src/lib/libhsm.c.orig
+++ libhsm/src/lib/libhsm.c
@@ -2174,10 +2174,12 @@ hsm_sign_buffer(hsm_ctx_t *ctx,
                                             CKM_GOSTR3411, digest_len,
                                             sign_buf);
             break;
-#if (LDNS_REVISION >= ((1<<16)|(7<<8)|(0)))
+#if USE_ED25519
         case LDNS_SIGN_ED25519:
             data_direct = 1;
             break;
+#endif
+#if USE_ED448
         case LDNS_SIGN_ED448:
             data_direct = 1;
             break;
@@ -2224,10 +2226,12 @@ hsm_sign_buffer(hsm_ctx_t *ctx,
         case LDNS_SIGN_ECDSAP384SHA384:
             sign_mechanism.mechanism = CKM_ECDSA;
             break;
-#if (LDNS_REVISION >= ((1<<16)|(7<<8)|(0)))
+#if USE_ED25519
         case LDNS_SIGN_ED25519:
             sign_mechanism.mechanism = CKM_EDDSA;
             break;
+#endif
+#if USE_ED448
         case LDNS_SIGN_ED448:
             sign_mechanism.mechanism = CKM_EDDSA;
             break;
